### PR TITLE
multi-host overlay driver quick start doc

### PR DESCRIPTION
**Pull request by nerdalert**

### CODE REVIEW

File: docs/overlay.md [hunk 1]
```diff
@@ -1,6 +1,146 @@
-Overlay Driver
```
Comment (mavenugo) [inline docs/overlay.md]: This document is supposed to carry the design details for overlay driver.
I prefer if you can add a new file named overlay-quckstart-config.md or something.

If you prefer to keep it in 1 file, Atleast I will keep the heading on `Overlay Driver` and also add a section for `Design` and mark it as TODO, followed by the quick start guide.

File: docs/overlay.md [hunk 2]
```diff
@@ -1,6 +1,146 @@
...
+### Start the Docker Daemon with the Network Driver Daemon Flags
+
+**host-1** Docker daemon:
+
+```
+$ docker -d --kv-store=consul:localhost:8500 --label=com.docker.network.driver.overlay.bind_interface=eth0
+```
+
+**host-2** Docker daemon needs to point to member of the Consul cluster. The following example starts the Docker daemon in the foreground. The global flags can be added to the docker configuration file:
+
```
Comment (mavenugo) [inline docs/overlay.md]: Please reduce the verbose texts.

File: docs/overlay.md [hunk 3]
```diff
@@ -1,6 +1,146 @@
...
+```
+$ docker -d --kv-store=consul:localhost:8500 --label=com.docker.network.driver.overlay.bind_interface=eth0 --label=com.docker.network.driver.overlay.neighbor_ip=<host-1-ip-address>
+```
+
+### QuickStart Containers Attached to a Network
+
+**host-1** Start a container on the network `dev` with a service name of `svc1`
```
Comment (mavenugo) [inline docs/overlay.md]: Would you mind replacing the above text with -

Start a container that publishes a service `svc1` in the network `dev` that is managed by `overlay` driver

File: docs/overlay.md [hunk 4]
```diff
@@ -1,6 +1,146 @@
...
+root@21578ff721a9:/# ip add show eth0
+34: eth0: <BROADCAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
+    link/ether 02:42:ec:41:35:bf brd ff:ff:ff:ff:ff:ff
+    inet 172.21.0.16/16 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::42:ecff:fe41:35bf/64 scope link
+       valid_lft forever preferred_lft forever
+```
+
+**host-2** Start a container on the network `dev` with a service name of `svc2`
```
Comment (mavenugo) [inline docs/overlay.md]: Same here

File: docs/overlay.md [hunk 5]
```diff
@@ -1,6 +1,146 @@
...
+
+```
+$ docker run -i -t --publish-service=svc2.dev.overlay debian
+root@d217828eb876:/# ping svc1
+PING svc1 (172.21.0.16): 56 data bytes
+64 bytes from 172.21.0.16: icmp_seq=0 ttl=64 time=0.706 ms
+64 bytes from 172.21.0.16: icmp_seq=1 ttl=64 time=0.687 ms
+64 bytes from 172.21.0.16: icmp_seq=2 ttl=64 time=0.841 ms
+```
+### Verbose Setup
```
Comment (mavenugo) [inline docs/overlay.md]: Detailed setup

File: docs/overlay.md [hunk 6]
```diff
@@ -1,6 +1,146 @@
...
+
+Once a container is started, a container on `host-1` and `host-2` both containers should be able to ping one another via IP, service name, \<service name>.\<network name>
+
+
+View information about the networks and services using `ls` and `info` subcommands like so:
+
+```
+$ docker service ls
+SERVICE ID          NAME                  NETWORK             CONTAINER
+0771deb5f84b        db2                   prod
```
Comment (mavenugo) [inline docs/overlay.md]: As per the flow the CONTAINER column cannot be empty for db2

File: docs/overlay.md [hunk 7]
```diff
@@ -1,6 +1,146 @@
...
+	Name: prod
+	Type: overlay
+
+$ docker service info db1.prod
+Service Id: aea23b224acfd2da9b893870e0d632499188a1a4b3881515ba042928a9d3f465
+	Name: db1
+	Network: prod
+```
+
+To detach and unpublish a service, the use the following syntax:
```
Comment (mavenugo) [inline docs/overlay.md]: To detach a container and unpublish a service

File: docs/overlay.md [hunk 8]
```diff
@@ -1,6 +1,146 @@
...
+
+```
+$ docker service detach $cid <service>.<network>
+$ docker service unpublish <service>.<network>
+```
```
Comment (mavenugo) [inline docs/overlay.md]: Please add examples.

File: docs/overlay.md (hunk 1 above)
Comment (nerdalert) [inline docs/overlay.md]: Done, thanks.

File: docs/overlay.md [hunk 9]
```diff
@@ -1,6 +1,154 @@
-Overlay Driver
-==============
+# Overlay Driver
 
-## Configuration
+### Multi-Host Overlay Driver Quick Start
 
-## Usage
+This example is to provision two Docker Hosts with the **experimental** Libnetwork default overlay network driver.
```
Comment (mavenugo) [inline docs/overlay.md]: overlay is not the default driver atm.

File: docs/overlay.md [hunk 10]
```diff
@@ -1,6 +1,154 @@
-Overlay Driver
-==============
+# Overlay Driver
```
Comment (mavenugo) [inline docs/overlay.md]: Thanks. Can you please add a Design section and just add a TODO ?

File: docs/overlay.md [hunk 11]
```diff
@@ -1,6 +1,154 @@
...
-## Configuration
+### Multi-Host Overlay Driver Quick Start
 
-## Usage
+This example is to provision two Docker Hosts with the **experimental** Libnetwork default overlay network driver.
+
+### Pre-Requisites
+
+- Kernel >= 3.1.6
+- Install Docker Experimental
```
Comment (mavenugo) [inline docs/overlay.md]: I dont think installing docker experimental is a prereq. Its part of the procedure that you have listed down

File: docs/overlay.md [hunk 12]
```diff
@@ -1,6 +1,154 @@
...
+```
+$ curl -OL https://dl.bintray.com/mitchellh/consul/#0.5.2_linux_amd64.zip
+$ unzip 0.5.2_linux_amd64.zip
+$ mv consul /usr/local/bin/
+```
+
+**host-1** Start Consul as a server in bootstrap mode:
+
+``` 
+$ consul agent -server -bootstrap -data-dir /tmp/consul -bind=<host-1-ip-address>`
```
Comment (mavenugo) [inline docs/overlay.md]: Can you remove ` at the end of the above line ?

File: docs/overlay.md [hunk 13]
```diff
@@ -1,6 +1,154 @@
...
+
+Once a container is started, a container on `host-1` and `host-2` both containers should be able to ping one another via IP, service name, \<service name>.\<network name>
+
+
+View information about the networks and services using `ls` and `info` subcommands like so:
+
+```
+$ docker service ls
+SERVICE ID          NAME                  NETWORK             CONTAINER
+0771deb5f84b        db2                   prod				  0e54a527f22c
```
Comment (mavenugo) [inline docs/overlay.md]: alignment issue on the CONTAINER column

File: docs/overlay.md (hunk 10 above)
Comment (nerdalert) [inline docs/overlay.md]: Added.

**Comment by mavenugo** [comment]
Thanks Brent. LGTM.

**Comment by mrjana** [comment]
LGTM

**Pull request merged**